external/concealer/include/concealer/utility/service_with_validation.h

```diff
@@ -0,0 +1,97 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CONCEALER__SERVICE_WITH_VALIDATION_HPP_
+#define CONCEALER__SERVICE_WITH_VALIDATION_HPP_
+
+#include <chrono>
+#include <string>
+
+#include "concealer/autoware.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace concealer
+{
+template <class T>
+class ServiceWithValidation
+{
+public:
+  ServiceWithValidation(std::string serviceName, Autoware & autoware)
```

Comment:
The constructor should be explicit.
Even if it seems more convenient to be implicit, it is often detrimental in the long run.

Reply:
Also, if there is no reason, do not copy the argument serviceName and use const reference to receive it.